### PR TITLE
Add k8s-app label for the kube-state-metrics ServiceMonitor

### PR DESCRIPTION
**Pull request by rajatvig**

### CODE REVIEW

File: jsonnet/kube-prometheus/kube-state-metrics/kube-state-metrics.libsonnet [hunk 1]
```diff
@@ -52,6 +52,7 @@
                               labels: {
                                 'app.kubernetes.io/name': 'kube-state-metrics',
                                 'app.kubernetes.io/version': ksm.version,
+                                'k8s-app': 'kube-state-metrics',
```
Comment (lilic) [inline jsonnet/kube-prometheus/kube-state-metrics/kube-state-metrics.libsonnet]: What is the need for adding this, we already have `app.kubernetes.io/name`which is the official recommended label https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels

Comment (rajatvig) [reply jsonnet/kube-prometheus/kube-state-metrics/kube-state-metrics.libsonnet]: The `k8s-app` label is applied to all other service monitors in this repository. This just makes it consistent with everything else.

Comment (lilic) [reply jsonnet/kube-prometheus/kube-state-metrics/kube-state-metrics.libsonnet]: Okay thanks, makes sense
cc @brancz any objections?

Comment (brancz) [reply jsonnet/kube-prometheus/kube-state-metrics/kube-state-metrics.libsonnet]: It's not a standardized label, I don't think we should continue using it.

Comment (rajatvig) [reply jsonnet/kube-prometheus/kube-state-metrics/kube-state-metrics.libsonnet]: @brancz Our target discovery has ended up relying on that label but I see the point you are making.
Would it make sense then to remove all the other instances and allow it to be overridden?
Can rework the PR to that effect.

Comment (brancz) [reply jsonnet/kube-prometheus/kube-state-metrics/kube-state-metrics.libsonnet]: I think we can work on each component one by one to move to the recommended app labels yes. I would prefer if we can do that one by one for each component one PR for each.

Comment (rajatvig) [reply jsonnet/kube-prometheus/kube-state-metrics/kube-state-metrics.libsonnet]: 👍

**Comment by rajatvig** [comment]
@brancz @lilic Here is the first two PRs
1. https://github.com/coreos/kube-prometheus/pull/458
2. https://github.com/coreos/kube-prometheus/pull/459

**Pull request closed without merging**